include/nestegg/nestegg.h

```diff
@@ -439,6 +452,18 @@ int nestegg_packet_encryption(nestegg_packet * packet);
 int nestegg_packet_iv(nestegg_packet * packet, unsigned char const ** iv,
                       size_t * length);
 
+/** Query the packet for offsets.
+@param packet            Packet initialized by #nestegg_read_packet.
+@param partition_offsets Storage for queried offsets.
+@param num_offsets       Length of returned offsets, may be 0.
```

Comment:
It can only be 0 if the packet isn't partitioned, right?  In that case, it seems like it'd be better to return an error instead and require the caller to have checked `nestegg_packet_partitioned` (or `_encrypted` if we roll them together) to verify the packet is partitioned before trying to query the partition offsets.